app/filters/dates.js

```diff
@@ -162,113 +164,89 @@ filters.prettyMonth = (monthNumber) => {
-
-
-    array.sort((a, b) => {
-      let x = (attr) ? a[attr] : a
-      let y = (attr) ? b[attr] : b
-
-      // Convert arrays of 3 to date objects
-      x = (_.isArray(x) && (x.length == 3)) ? filters.arrayToDateObject(x) : x
-      y = (_.isArray(y) && (y.length == 3)) ? filters.arrayToDateObject(y) : y
-
-      // console.log({x}, {attr})
-      // if (!caseSens && _.isString(x) && _.isString(y)) {
-      //   x = x.toLowerCase()
-      //   y = y.toLowerCase()
-      // }
-
-      if (x < y) {
-        return reversed ? 1 : -1
-      } else if (x > y) {
-        return reversed ? -1 : 1
-      } else {
-        return 0
-      }
-    })
-
-    return array
-  }
+  return array
+}
 
 // https://momentjs.com/docs/#/displaying/format/
 filters.formatDate = (date, format, dateFormat) => {
 
-  var returnDate;
+  let returnDate
   // No date provided.
-  if (!date){
-    // console.log('error for', date, 'format', format);
-    return ''
-    // throw "Error in formatDate: no date provided";
+  if (!date) {
+    // console.log('error for', date, 'format', format)
+    return null
+    // throw "Error in formatDate: no date provided"
   }
   // Check for valid date
-  else if (dateFormat && moment(date, dateFormat).isValid()){
-    returnDate = moment(date, dateFormat);
+  else if (_.isString(dateFormat) && DateTime.fromISO(date).isValid) {
+    // returnDate = moment(date, dateFormat)
+    returnDate = DateTime.fromISO(date).toFormat(dateFormat)
   }
-  else if ( moment(date).isValid() ){
-    returnDate = moment(date);
+  else if ( DateTime.fromISO(date).isValid ) {
+    returnDate = DateTime.fromISO(date)
   }
   // Invalid date
   else {
-    throw "Error in formatDate: invalid date";
-  };
-
-  switch (true)
-    {
-      // 2018-03-21
-      case (format == 'dashDate'):
-        return returnDate.format('YYYY-MM-DD');
-
-      // 2018/03/21
-      case (format == 'slashDate'):
-        return returnDate.format('YYYY/MM/DD');
-
-      // 2018/03
-      case (format == 'yearMonth'):
-        return returnDate.format('YYYY/MM');
-
-      // 2018-03-21T00:00:00.000Z
-      case (format == 'iso8601'):
-        return returnDate.toISOString();
-
-      // a year ago
-      case (format == 'relative'):
-        return moment(returnDate).fromNow()
-        // return timeAgoInDays(returnDate)
-
-      // 21st March 2018
-      case (format == 'pretty'):
-        return returnDate.format('Do MMMM YYYY');
-
-      // March 21st 2018, 12:00:00 am
-      case (format == 'full'):
-        return returnDate.format('MMMM Do YYYY, h:mm:ss a');
-
-      // pass format through to moment
-      case _.isString(format):
-        return returnDate.format(format);
-
-      // Default
-      default:
-        return returnDate.format();
-    }
-}
+    throw "Error in formatDate: invalid date"
+  }
 
-filters.isInLast = (date, count, units) => {
-  let compareDate = moment().subtract(count, units)
-  return moment(date).isAfter(compareDate)
-}
+  switch (true) {
```

Comment:
Instead of switch(true) pass in format and then u don't have to check format == all the time?

Reply:
This is what I was going to do except the switch statement has mixed expressions, so I erred on the side of caution and decided not to refactor the whole statement. I can refactor the switch statement if that's your recommendation?